solver/src/interactions/uniswap.rs

```diff
@@ -92,7 +93,7 @@ mod tests {
         // Verify Approve
         let approve_call = &interactions[0];
         assert_eq!(approve_call.0, token_in);
-        let call = &approve_call.2;
+        let call = &approve_call.2 .0;
```

Comment:
Is the extra space here intentional!
```suggestion
        let call = &approve_call.2.0;
```

Reply:
Looks like rustfmt does it this way so I would just keep it. Apparently the rust compiler used to need the extra space (but not longer does) and rustfmt hasn't been updated to reflect that.

Reply:
AFAIU its because of the nested tuple structs:
```rust
pub struct Foo(Bar);
pub struct Bar(u64);

let foo: Foo = ...;
foo.0.0;
```

The issue is parsing ambiguity on `foo.0.0`:
```
identifier(foo) dot(.) tuple_field(0) dot(.) tuple_field(0)
```
or
```
identifier(foo) dot(.) float(0.0)
```